src/Juvix/Compiler/Backend/Geb/Translation/FromCore.hs

```diff
@@ -414,17 +416,33 @@ fromCore tab = case tab ^. Core.infoMain of
                 _ -> impossible
           [] -> impossible
 
-        mkBranch :: Object -> Core.Node -> Core.CaseBranch -> Trans Morphism
+        mkBranch :: Object -> Morphism -> Core.CaseBranch -> Trans Morphism
         mkBranch valty val Core.CaseBranch {..} = do
           branch <- underBinders _caseBranchBindersNum (convertNode _caseBranchBody)
           if
               | _caseBranchBindersNum == 0 -> return branch
-              | otherwise -> do
-                  val' <- convertNode val
-                  return $ mkApps (mkLambs branch argtys) val' valty argtys
+              | _caseBranchBindersNum == 1 ->
+                  return $
+                    MorphismApplication
+                      Application
+                        { _applicationDomainType = valty,
```

Comment:
can you keep the pattern, here it'd be: "valType" or "domainType".

Reply:
Done